src/Sql/Sql/ServiceObjective/Service/AzureSqlServerServiceObjectiveAdapter.cs

```diff
@@ -55,10 +62,16 @@ public AzureSqlServerServiceObjectiveAdapter(IAzureContext context)
         /// <param name="serverName">The name of the server</param>
         /// <param name="serviceObjectiveName">The name of the service objective</param>
         /// <returns>The ServiceObjective</returns>
-        public AzureSqlServerServiceObjectiveModel GetServiceObjective(string resourceGroupName, string serverName, string serviceObjectiveName)
+        public List<AzureSqlServerServiceObjectiveModel> GetServiceObjective(string resourceGroupName, string serverName, string serviceObjectiveName)
         {
-            var resp = Communicator.Get(resourceGroupName, serverName, serviceObjectiveName);
-            return CreateServiceObjectiveModelFromResponse(resourceGroupName, serverName, resp);
+            var server = ServerCommunicator.Get(resourceGroupName, serverName);
+            var capabilities = CapabilitiesCommunicator.Get(server.Location);
+
+            return (
+                from sv in FilterByName(capabilities.SupportedServerVersions, server.Version)
+                from e in sv.SupportedEditions
+                from slo in FilterByName(e.SupportedServiceLevelObjectives, serviceObjectiveName)
+                select CreateServiceObjectiveModelFromResponse(e, slo, resourceGroupName, serverName)).ToList();
```

Comment:
It should be a singleton returned, but it seems easier to just return list rather than convert to scalar here and then back to list at call site.